modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -1029,15 +1085,22 @@ public void setEmbeddedScene(EmbeddedSceneInterface embeddedScene) {
                 return;
             }
             if (pWidth > 0 && pHeight > 0) {
-                scenePeer.setSize(pWidth, pHeight);
+                lScenePeer.setSize(pWidth, pHeight);
+            }
+            lScenePeer.setPixelScaleFactors((float) scaleFactorX, (float) scaleFactorY);
+            synchronized(LOCK) {
+                scenePeer = lScenePeer;
```

Comment:
I wonder if the whole method should be synchronized?
the logic around scenePeer is asking to be atomic, isn't it?